### ISSUE TITLE
Possible bug: Nyxt unexpectadly closes buffer on the dockerized version of Nyxt for macOS

### ISSUE THREAD

**Issue by pdelfino** [deddu/nyxt-docker#8]
**Describe the bug**

This video describes what was written in the [title](https://www.youtube.com/watch?v=LJaTL0_uTkc&t=12s).

**Precise recipe to reproduce the issue**

1 - open Nyxt following the description on [Readme](https://github.com/deddu/nyxt-docker#usage-for-docker-on-macos) "usage for Docker on macOS". Except for this step "Launch a new xterm terminal and ensure you have “DISPLAY=:0” set." which I did not do because I was not sure about how to do it.
2 - open [this](https://www.folha.uol.com.br/) website. This is something like the Brazilian version of New York Times
3 - Wait. Suddenly, Nyxt will close the buffer
4 - The mini buffer will send an error message which can be read during the video with the bug description

The same problem also happened with my [blog](http://www.pdelfino.com.br/)'s website. This problem **does not** happen when browsing stuff like `manual`, `tutorial` (kind of internal buffers). Both "problematic websites" have always worked fine while using Nyxt on my Linux machine.

**Information**
- OS name+version: macOS 12
- Graphics card and driver: (I will send soon)
- Desktop environment / Window manager name+version: default Desktop Environment on macOS
- How you installed Nyxt (Guix pack, package manager, build from source): Docker
- Information from `show-system-information`: (I will send soon)

Observation: this bug did not happen to me. I do not owe a mac. I am trying to help a friend who is craving to use Nyxt on his mac.

**Comment by deddu** [comment]
I did fiddle a bit with the packages, and I could get to render the youtube page without a crash. Also the video did start playing but it crashed after a bit - here's the output
```
<INFO> [00:51:02] Listening to socket "nyxt/nyxt.socket".
Nyxt version 2.2.2
<INFO> [00:51:10] Loading "https://youtube.com/".
<INFO> [00:51:15] Finished loading "https://www.youtube.com/".
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_card_driver retured error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_concat returned eror: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_refer returned eror: No such file or directory
ALSA lib conf.c:5220:(snd_config_expand) Evaluate error: No such file or directry
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM default
AL lib: (EE) ALCplaybackAlsa_open: Could not open playback device 'default': Nosuch file or directory

(WebKitWebProcess:187): Gdk-ERROR **: 00:51:49.683: The program 'WebKitWebProces' received an X Window System error.
This probably reflects a bug in the program.
The error was 'GLXBadRenderRequest'.
  (Details: serial 634 error_code 167 request_code 149 (GLX) minor_code 1)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
<WARN> [00:51:49] Warning: Web process crashed for buffer 2

```

now, for your question: there's no lspci, I'll install that and post an update.

**Comment by Ambrevar** [comment]
Try installing sound-related packages like alsa-lib, alsa-plugins*,
openal, pulseaudio.

You might also want to fiddle with /etc/asound.conf, but I don't know
the details.

**Comment by dre-sctg** [comment]
so, with regard to adding alsa
-  I did add a bunch of libs
-   [disabled dbus warnings](https://unix.stackexchange.com/questions/230238/x-applications-warn-couldnt-connect-to-accessibility-bus-on-stderr)
-  tried to follow [this SO on docker audio devices looks promising](https://stackoverflow.com/questions/41083436/how-to-play-sound-in-a-docker-container-on-mac-os-yosemite)
-   however sound is not exposed by default, and is not supported
-   but where's the will there's a way:
  -   [X] tried to run `docker ... --privileged --device /dev/snd ...` but there's no such device on mac
  -   [X] tried to `/dev/null` but no luck yet
  -   [ ] try  [exposing pulseaudio](https://stackoverflow.com/questions/40136606/how-to-expose-audio-from-docker-container-to-a-mac/50939994#50939994)
 -   [ ] try to output to a dummy device
    -   the kernel module is not available, so [this is a bit hard to follow](https://www.alsa-project.org/main/index.php/Matrix:Module-dummy)



here's the error
```
<INFO> [11:22:08] Loading "https://youtube.com/".
<INFO> [11:22:13] Finished loading "https://www.youtube.com/".
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5220:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM default
AL lib: (EE) ALCplaybackAlsa_open: Could not open playback device 'default': No such file or directory

(WebKitWebProcess:177): Gdk-ERROR **: 11:22:21.719: The program 'WebKitWebProcess' received an X Window System error.
This probably reflects a bug in the program.
The error was 'GLXBadRenderRequest'.
  (Details: serial 650 error_code 167 request_code 149 (GLX) minor_code 1)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
<WARN> [11:22:21] Warning: Web process crashed for buffer 2
^[Gdk-Message: 11:27:59.743: WebKitWebProcess: Fatal IO error 11 (Resource temporarily unavailable) on X server host.docker.internal:0.
```

**Comment by deddu** [comment]
I saw this one in my logs today ... perhaps worthy of further investigation. 

```
<WARN> [11:05:26] Failed to echo these args: ("Web process terminated for buffer ~a because ~[it crashed~;of memory exhaustion~;we had to close it~]"
                                              "2" :WEBKIT-WEB-PROCESS-CRASHED)
```

**Comment by Ambrevar** [comment]
This is fixed on master.
No worries, it'a harmless.
